arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,76 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+// const int frontPin = 0;
+// GP2Y0A02 sideFrontIR(arduinoRuntime,
+//                     frontPin); 
+
+const int backPin = 3;
+GP2Y0A02 sideBackIR(arduinoRuntime,
+                    backPin); 
+
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 2000;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+
+SimpleCar car(control);
+
+void setup()
+{
+    Serial.begin(9600);
+    Serial.setTimeout(200);
+}
+
+void loop()
+{
+    handleInput();
+    autoStop();
+    delay(100);
+   
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void handleInput()
+{
+    // handle serial input if there is any
+    if (Serial.available())
+    {
+        String input = Serial.readStringUntil('\n');
+        if (input.startsWith("m"))
+        {
+            int throttle = input.substring(1).toInt(); 
+            car.setSpeed(throttle);
+        }
+        else if (input.startsWith("t"))
+        {
+            int deg = input.substring(1).toInt();
+            car.setAngle(deg);
+        }
+        
+        
+    }
+   
 }
+
+void autoStop(){
+  
+  if(front.getDistance() < 1000 && front.getDistance() > 0){ 
```

Comment:
Isn't `1000` too much? The distance is in centimeters.

Reply:
This must be a typo, it should be a 100

Reply:
After the value is corrected, there's another issue. `getDistance` is called twice meaning that there are two _different_ measurements taking place. What happens if the sensor returns different values, e.g. 0 and 200?